driver-core/src/main/com/mongodb/MongoClientSettings.java

```diff
@@ -629,6 +648,16 @@ public UuidRepresentation getUuidRepresentation() {
         return uuidRepresentation;
     }
 
+    /**
+     * Gets the server API to use when sending commands to the server.
+     *
+     * @return the server API
+     * @since 4.3
+     */
+    public Optional<ServerApi> getServerApi() {
```

Comment:
Not sure whether we should start introduction Optional in Settings, as there are many other settings that can be null and a new paradigm might be confusing.

Reply:
I'm of the mindset that `Optional` should be used in these cases, and I'd think most developers should be comfortable with it, so I agree with the decision here.

Reply:
I'm unsure on this, and we probably are the only consumers of the value. To me if we only use it like so: `getServerApi().orElse(null)` then its not worth wrapping in an optional.

Also we dont consistently wrap `getServerApi` methods in optionals (ClusterBinding).

I'd argue the inconsistency also adds extra cognitive load to the code base.

Reply:
Ross persuaded me.  I've remove use of Optional here and in MongoClientOptions